WordPress/Classes/Utility/Editor/GutenbergSettings.swift

```diff
@@ -22,17 +23,38 @@ class GutenbergSettings {
 
     // MARK: Public accessors
 
-    func isGutenbergEnabled() -> Bool {
-        return database.object(forKey: gutenbergEditorEnabledKey) as? Bool ?? false
+    /// True if gutenberg editor is currently enabled
+    private(set) var isGutenbergEnabled: Bool {
+        get {
+            return database.bool(forKey: Key.enabled)
+        }
+        set(enabled) {
+            database.set(enabled, forKey: Key.enabled)
+        }
+    }
+
+    /// True if gutenberg editor has been enabled at least once
+    var wasGutenbergEnabledOnce: Bool {
+        return database.object(forKey: Key.enabled) != nil
+    }
+
+    func shouldAutoenableGutenberg(for post: AbstractPost) -> Bool {
+        return  post.containsGutenbergBlocks() && !wasGutenbergEnabledOnce
+    }
+
+    func setGutenbergEnabledIfNeeded() {
+        if isGutenbergEnabled == false {
```

Comment:
What do you think about getting rid of `toggleGutenberg()` and moving the analytics to the `set` accessor of the property? I find the "toggle" semantics awkward since we know her we want to enable it (if it's disabled), and on app settings we also know the specific value to be set from the UI.

Reply:
Sounds good to me 👍 
`.toogleGutenberg()` sounds practical for the switch, but since it holds the analytics logic, is not the best for other uses. I'll give it a try to remove it, or its dependency on tracks.